File: ChessClubBurgas64.Data/Models/Puzzle.cs
#nullable disable
using System.ComponentModel.DataAnnotations;

namespace ChessClubBurgas64.Data.Models
{
    public class Puzzle
    {
        [Key]
        public int Id { get; set; }
        public int Number { get; set; }
        public string Description { get; set; }
        public string Solution { get; set; }
        public int Points { get; set; }
        public string Difficulty { get; set; }
        public string ImageUrl { get; set; }
        public string ImageId { get; set; }
        public virtual Image Image { get; set; }
    }
}

File: ChessClubBurgas64.Web/ClientApp/src/app/models/puzzle.ts
import {Image} from "./image"

export interface Puzzle {
    id: number;
    title: string;
    description: string;
    solution: string;
    points: number;
    difficulty: string;
    imageUrl: string;
    image: Image
}

export class PuzzleFormValues
{
  id?: number = undefined;
  title: string = '';
  description: string = '';
  solution: string = '';
  points: number = 1;
  difficulty: string = '';
  imageUrl: string = '';
  
  constructor(puzzle?: PuzzleFormValues) {
    if (puzzle) {
      this.id = puzzle.id;
      this.title = puzzle.title;
      this.description = puzzle.description;
      this.solution = puzzle.solution;
      this.points = puzzle.points;
      this.difficulty = puzzle.difficulty;
      this.imageUrl = puzzle.imageUrl;
    }
  }
}

export class Puzzle implements Puzzle {
  constructor(init?: PuzzleFormValues) {
    Object.assign(this, init);
  }
}
